lib/spack/spack/fetch_strategy.py

```diff
@@ -792,11 +802,460 @@ def __str__(self):
         return "[go] %s" % self.url
 
 
+class GitRef(object):
+    """A wrapper that knows how to format git refspecs in multiple contexts."""
+    _known_types = ['branch', 'commit', 'tag']  # type: ClassVar[List[str]]
+    _ref_type = None                            # type: str
+    _ref = None                                 # type: str
+
+    @classmethod
+    def known_types(cls):
+        # type: () -> List[str]
+        return cls._known_types
+
+    @property
+    def ref_type(self):
+        # type: () -> str
+        return self._ref_type
+
+    def __init__(self, ref_type, ref):
+        # type: (str, str) -> None
+        if ref_type not in self.known_types():
```

Comment:
Any reason that self.known_types is a method instead of a property (and then again why not just have this be a variable without the extra method or property?

Reply:
Fixed as per above!